pkg/kernelmonitor/kernel_log_watcher.go

```diff
@@ -144,14 +157,54 @@ func (k *kernelLogWatcher) watchLoop() {
 				continue
 			}
 			k.logCh <- log
-		case <-k.tomb.Stopping():
-			k.tl.Stop()
-			glog.Infof("Stop watching kernel log")
-			return
+		} else { // err == io.EOF
+			buffer.WriteString(line)
 		}
 	}
 }
 
+// getLogReader gets a kernel log reader.
+func getLogReader(path string) (io.Reader, error) {
+	reader, err := tryJournal()
+	if err == nil {
+		return reader, nil
+	}
+	reader, err = tryLogFile(path)
+	if err == nil {
+		return reader, nil
+	}
+	return nil, err
+}
+
+func tryJournal() (io.Reader, error) {
+	r, err := sdjournal.NewJournalReader(sdjournal.JournalReaderConfig{
+		NumFromTail: uint64(0),
+		Matches: []sdjournal.Match{
+			{
+				Field: sdjournal.SD_JOURNAL_FIELD_TRANSPORT,
+				Value: "kernel",
+			},
+		},
+	})
+	if err != nil {
+		return nil, fmt.Errorf("Error opening journal: %v", err)
+	}
+	if r == nil {
+		return nil, fmt.Errorf("Got a nil reader")
```

Comment:
Error should start with small letter. `got a nil reader` and same with the errors above.